atomics/include/desul/atomics/Macros.hpp

```diff
@@ -11,14 +11,21 @@ SPDX-License-Identifier: (BSD-3-Clause)
 
 // Macros
 
+// ONLY use GNUC atomics if not compiling for the device
+// and we didn't explicitly say to use OPENMP or SERIAL atomics
 #if defined(__GNUC__) && (!defined(__CUDA_ARCH__) || !defined(__NVCC__)) &&  \
     (!defined(__HIP_DEVICE_COMPILE) || !defined(__HIP_PLATFORM_HCC__)) &&    \
     !defined(__SYCL_DEVICE_ONLY__) && !defined(DESUL_HAVE_OPENMP_ATOMICS) && \
     !defined(DESUL_HAVE_SERIAL_ATOMICS)
 #define DESUL_HAVE_GCC_ATOMICS
 #endif
 
-#ifdef _MSC_VER
+// Equivalent to above: if we are compiling for the device we
+// need to use CUDA/HIP/SYCL atomics instead of MSVC atomics
+#if defined(_MSC_VER)  && \
+    (!defined(__CUDA_ARCH__) || !defined(__NVCC__)) && \
+    (!defined(__HIP_DEVICE_COMPILE) || !defined(__HIP_PLATFORM_HCC__)) && \
+    !defined(SYCL_LANGUAGE_VERSION)
```

Comment:
The version above uses `__SYCL_DEVICE_ONLY__`.  Did you mean to use `SYCL_LANGUAGE_VERSION` here?